api/storage_service.cc

```diff
@@ -220,30 +220,39 @@ seastar::future<json::json_return_type> run_toppartitions_query(db::toppartition
     });
 }
 
-future<json::json_return_type> set_tables_autocompaction(http_context& ctx, const sstring &keyspace, std::vector<sstring> tables, bool enabled) {
+static future<json::json_return_type> set_tables(http_context& ctx, const sstring& keyspace, std::vector<sstring> tables, std::function<future<>(replica::table&)> set) {
     if (tables.empty()) {
         tables = map_keys(ctx.db.local().find_keyspace(keyspace).metadata().get()->cf_meta_data());
     }
 
+    return do_with(keyspace, std::move(tables), [&ctx, set] (const sstring &keyspace, const std::vector<sstring>& tables) {
+            // FIXME: restore indentation.
```

Comment:
nit: `set` is captured instead of `enabled` so this whole block appears in the patch hunk.
No need to keep indentation....

Reply:
also, can just be converted to a coroutine to simplify

Reply:
I can convert it into coroutine after all the changes, otherwise it will make our lifes harder when backporting